game/engine/persistent_server.py

```
@@ -0,0 +1,142 @@
+
+    proc = None
+    
+    def start_engine():
+        nonlocal proc
+        if proc is not None:
+            try:
+                proc.terminate()
+                proc.wait(timeout=1)
+            except Exception:
+                pass
+        proc = subprocess.Popen(
```

Ruff (0.15.20)

[error] `subprocess` call: check for execution of untrusted input

(S603)